tests/signal_manager_test.sh

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+
+include './src/signal_manager.sh'
+include './tests/utils'
+
+function test_default_handler()
+{
+  local -r original_dir="$PWD"
+  local -r expected=$'\n'"Oh no! An interruption! See ya..."
+
+  cd "$SHUNIT_TMPDIR"
+
+  signal_manager 'kill -s SIGTERM $!' SIGUSR1
+  {
+    signal_manager default_interrupt_handler SIGINT SIGTERM
+    sleep infinity &
+    kill -s SIGUSR1 $$ # Send SIGUSR1 to parent process
+    wait $! # wait for SIGTERM
+  } > out.tmp &
+
+  wait $! # wait for SIGUSR1
+  assertEquals "($LINENO)" "$expected" "$(cat out.tmp)"
+
+  rm -f out.tmp
+
+  signal_manager 'kill -s SIGINT $!' SIGUSR1
+  {
+    signal_manager default_interrupt_handler SIGINT SIGTERM
+    sleep infinity &
+    kill -s SIGUSR1 $$ # Send SIGUSR1 to parent process
+    wait $! # wait for SIGINT
+  } > out.tmp &
+
+  wait $!
+  assertEquals "($LINENO)" "$expected" "$(cat out.tmp)"
+
+  killall sleep
```

Comment:
What happens if I have a sleep command running while I execute this test? I'm wondering if `killall` will not kill other `sleep` in the system. Maybe we should use the PID instead of killing all `sleep`.